applications/dashboard/controllers/api/MediaApiController.php

```diff
@@ -292,15 +304,36 @@ public function normalizeOutput(array $row) {
     public function post(array $body) {
         $this->permission('Garden.Uploads.Add');
 
+        switch ($body['type']) {
+            case self::TYPE_TEXT:
+                $uploadExtensions = self::EXTENSIONS_TEXT;
+                break;
+            case self::TYPE_BINARY:
+                $uploadExtensions = self::EXTENSIONS_BINARY;
+                break;
+            case self::TYPE_ARCHIVE:
+                $uploadExtensions = self::EXTENSIONS_ARCHIVE;
+                break;
+            case self::TYPE_IMAGE:
+            default:
+                $uploadExtensions = array_keys(ImageResizer::getExtType());
+        }
+
         $uploadSchema = new UploadedFileSchema([
-            'allowedExtensions' => array_keys(ImageResizer::getExtType())
+            'allowedExtensions' => $uploadExtensions
         ]);
 
         $in = $this->schema([
             'file' => $uploadSchema,
             'type:s' => [
                 'description' => 'The upload type.',
-                'enum' => ['image']
+                'default' => self::TYPE_IMAGE,
```

Comment:
It would make more sense to me for the client to be required to tell us what type they're uploading, instead of making an assumption if they don't specify.